folly/CMakeLists.txt

```diff
@@ -148,6 +154,17 @@ target_include_directories(folly PUBLIC ${FOLLY_ROOT})
 
 target_link_libraries(folly boost)
 
+# The "-lgcc_s" added before "-lunwind" is used to solve a linking
+# order issue that causes "_Ux86_64_setcontext" to be called from
+# libunwind instead of libgcc_s and messes up the signal mask
+# (more details here http://savannah.nongnu.org/bugs/?48486). The
+# libgcc_s dependency is already part of HHVM, this enforces the
+# order in which the libraries are linked.
+if (ENABLE_FOLLY_SYMBOLIZER)
+  SET(UNWIND_LIB -lgcc_s -lunwind)
```

Comment:
Do you know what happens on mac (clang) with and without this?

Reply:
I have not tested on either mac or clang, which is why I protected this option under GCC in the related PR in HHVM: https://github.com/facebook/hhvm/pull/8123

Do you think it is better to include the GCC requirement in folly as well?

Reply:
We don't actually use Folly's build system - but given that https://github.com/facebook/folly/blob/9656bacc814ce9efd24e178ed5e127c5c365a248/CMake/FollyConfigChecks.cmake checks for elf.h , mac's most likely not going to be an issue - but Clang should work.

cc @simpkins

Reply:
@fredemmott can you please clarify if I should change anything for Clang?